airflow/providers/google/cloud/operators/cloud_storage_transfer_service.py

```diff
@@ -546,28 +547,29 @@ def __init__(
     ) -> None:
         # To preserve backward compatibility
         # TODO: remove one day
```

Comment:
I'm pretty sure that this fallback broken for a long time, this part exists before this code migrated from the `airflow.contrib` to the `providers` in https://github.com/apache/airflow/pull/6077 I do not trace this change back.

`filter` parameter keep into the `kwargs` and pass to subclass and this will raise an error that parameter not defined.

So another option just remove entire fallback block, since it do not work well for more than 3-4 years.

Reply:
Remove it.

Reply:
Move this removal into the separate PR